packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts

```diff
@@ -176,11 +176,13 @@ export class SnowflakeWarehouseClient implements WarehouseClient {
         } catch (e) {
             throw new WarehouseQueryError(e.message);
         } finally {
-            // todo: does this need to be promisified? uncaught error in callback?
-            connection.destroy((err) => {
-                if (err) {
-                    throw new WarehouseConnectionError(err.message);
-                }
+            await new Promise((resolve, reject) => {
```

Comment:
Nice!! There is a native promisify if you wanted to, I don't mind : https://nodejs.org/api/util.html#utilpromisifyoriginal

Reply:
I guess you'd then have to wrap in a try/catch to customise the error type